### PR TITLE
feat: 🆕  add support use dots to provide dotsClasses

### LINKED ISSUE #21131: Not able to forward dotClass prop in carousel
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/polished-morning-lb2qm)

### Steps to reproduce
1) Pass custom dot class name into a carousel.
2) Not able to forward that class name
3) " const dotsClass = 'slick-dots';
    const dotPosition = this.getDotPosition();
    props.vertical = dotPosition === 'left' || dotPosition === 'right';
    props.dotsClass = `${dotsClass} ${dotsClass}-${dotPosition || 'bottom'}`;"

### What is expected?
1) " const dotsClass = 'slick-dots';
    const dotPosition = this.getDotPosition();
    props.vertical = dotPosition === 'left' || dotPosition === 'right';
    props.dotsClass = `${dotsClass} ${dotsClass}-${dotPosition || 'bottom'} ${props.dotsClass}`;"

### What is actually happening?
Not able to forward a custom dot class name

| Environment | Info |
|---|---|
| antd | 3.26.5 |
| React | 16.3.0 |
| System | Mac |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by afc163**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
same as #21156 wants to do
close #21131 
close #21156

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |    dots支持对象传递dotsClass       |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/carousel/index.en-US.md](https://github.com/ant-design/ant-design/blob/DotClass-string/components/carousel/index.en-US.md)
[View rendered components/carousel/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/DotClass-string/components/carousel/index.zh-CN.md)

-----
[View rendered components/carousel/index.en-US.md](https://github.com/ant-design/ant-design/blob/DotClass-string/components/carousel/index.en-US.md)
[View rendered components/carousel/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/DotClass-string/components/carousel/index.zh-CN.md)

### CODE REVIEW

**Comment by afc163** [comment]
approve needed

File: components/carousel/__tests__/index.test.js [hunk 1]
```diff
@@ -126,4 +126,18 @@ describe('Carousel', () => {
       ).toBeTruthy();
     });
   });
+
+  describe('dots precise control by plain object', () => {
+    it('use dots to provide dotsClasse', () => {
+      const wrapper = mount(
+        <Carousel dots={{ dotsClass: 'customDots' }}>
```
Comment (afc163) [inline components/carousel/__tests__/index.test.js]: https://github.com/ant-design/ant-design/pull/21846#issuecomment-594334083

@yoyo837

**Comment by afc163** [comment]
冲突解决一下。

**Comment by afc163** [comment]
我先合并一下 feature https://github.com/ant-design/ant-design/pull/21849

**Review by yoyo837 — APPROVED** [review]

**Comment by afc163** [comment]
目测好了

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**